### PR TITLE
Setup Cloudinary for profile picture uploading

### PR DESCRIPTION
**Pull request by serafy0**
### What this PR does (required):
- I tried to setup Cloudinary
- made changes to the user model by embedding a sub-document for the profile picture that contains the Cloudinary public id and url
- added routes that allows the user to upload their profile photo and integrated it in the front-end.

### Screenshots / Videos (required):
Screen Shot 2021-08-26 at 12 30 56 AM

### Any information needed to test this feature (required):
- the same as the ticket before. login in as a user then click on the user's profile picture in the top left, then you should be able to delete or add a new profile picture and the changes should persist

### Any issues with the current functionality (optional):
- I'm not sure if embedding the image property inside the user is correct. or if I should create a separate model for Images
- does this also cover [issue#37](https://github.com/hatchways/team-naan-bread/issues/37) ?

### CODE REVIEW

File: client/src/pages/Profile/ProfilePhoto/ProfilePhoto.tsx [hunk 1]
```diff
@@ -19,20 +27,28 @@ export default function ProfilePhoto(): JSX.Element {
     },
   });
 
+  useEffect(() => {
+    setPhoto(loggedInUser.profile_photo_url);
+  }, [loggedInUser.profile_photo_url]);
```
Comment (moffatethan) [inline client/src/pages/Profile/ProfilePhoto/ProfilePhoto.tsx:32]: Going forward, I would suggest to use camelCase as opposed to snake_case. This is JavaScript convention

File: server/controllers/user.js [hunk 2]
```diff
@@ -21,3 +22,37 @@ exports.searchUsers = asyncHandler(async (req, res, next) => {
...
+  await cloudinary.uploader.destroy(user.profile_photo.public_id);
+  user.profile_photo = null;
+};
+
+exports.uploadProfilePhoto = asyncHandler(async (req, res, next) => {
+  const image = req.file;
+  const user = await User.findById(req.user.id);
+  if (!user) {
+    return res.sendStatus(403);
+  }
```
Comment (moffatethan) [inline server/controllers/user.js:36]: 403 really isn't the right status code for this kind of response. The user couldn't be found so a 404 is more appropriate and follows REST conventions more.

File: server/controllers/user.js [hunk 3]
```diff
@@ -21,3 +22,37 @@ exports.searchUsers = asyncHandler(async (req, res, next) => {
...
+  if (user.profile_photo.url) {
+    await deletePhoto(user);
+  }
+  user.profile_photo = {
```
Comment (moffatethan) [inline server/controllers/user.js:40]: camelCase here please. 👍🏻

File: server/controllers/user.js [hunk 4]
```diff
@@ -21,3 +22,37 @@ exports.searchUsers = asyncHandler(async (req, res, next) => {
...
+    url: image.path,
+    public_id: image.filename,
```
Comment (moffatethan) [inline server/controllers/user.js:42]: and here

File: server/controllers/user.js [hunk 5]
```diff
@@ -21,3 +22,37 @@ exports.searchUsers = asyncHandler(async (req, res, next) => {
...
+  };
+  user.save();
+
+  res.sendStatus(200);
+});
+exports.deleteProfilePhoto = asyncHandler(async (req, res, next) => {
+  const user = await User.findById(req.user.id);
+  if (!user) {
+    res.sendStatus(403);
```
Comment (moffatethan) [inline server/controllers/user.js:51]: 404 error

**Review by moffatethan — APPROVED** [review]
Good improvement thank you!

**Comment by serafy0** [comment]
I've tried making some changes. is it better if i push it to my own `moBranch` or to `main` ?

**Pull request merged**